web_src/js/features/formatting.js

```diff
@@ -13,9 +16,34 @@ export function initFormattingReplacements() {
     }
   }
 
-  // for each <time></time> tag, if it has the data-format="date" attribute, format
-  // the text according to the user's chosen locale
-  for (const timeElement of document.querySelectorAll('time[data-format="date"]')) {
-    timeElement.textContent = dateFormatter.format(new Date(timeElement.dateTime));
+  // for each <time></time> tag, if it has the data-forma attribute, format
+  // the text according to the user's chosen locale and formatter
+  formatAllTimeElements();
+}
+
+function formatAllTimeElements() {
+  const formats = ['date', 'short-date', 'date-time'];
+  for (const f of formats) {
+    formatTimeElements(f);
+  }
+}
+
+function formatTimeElements(format) {
+  let formatter;
+  switch (format) {
+    case 'date':
+      formatter = dateFormatter;
+      break;
+    case 'short-date':
+      formatter = shortDateFormatter;
+      break;
+    case 'date-time':
+      formatter = dateTimeFormatter;
+      break;
+    default:
+      throw new Error('Unknown format');
+  }
+  for (const timeElement of document.querySelectorAll(`time[data-format="${format}"]`)) {
```

Comment:
Use `document.querySelectorAll("time[data-format]")` to get all elements in one shot.

Reply:
Right, I don't want to get all in one shot

Reply:
Why not? If you take a test, I think `document.querySelectorAll("time[data-format]") ` could be faster.

Reply:
Ok, done